src/moscot/solvers/_output.py

```diff
@@ -167,6 +177,79 @@ def chain(self, outputs: Iterable["BaseSolverOutput"], scale_by_marginals: bool
+        self,
+        mode: Literal["threshold", "percentile", "min_1"] = SparsifyMode.THRESHOLD,
+        threshold: float = 1e-8,
+        batch_size: int = 1024,
+        n_samples: Optional[int] = None,
+    ) -> None:
+        """
+        Sparsify the transport matrix.
+
+        This function sets all entries of the transport matrix below `threshold` to 0 and
+        returns the result as a :class:`scipy.sparse.csr_matrix`.
+
+        Parameters
+        ----------
+        mode
+            Which threshold to use for sparsification. Can be one of:
+
+                - "threshold" - threshold below which entries are set to 0.0.
+                - "percentile" - determine threshold by percentile below which entries are set to 0. Hence, between 0
+                  and 100.
+                - "min_1" - choose the threshold such that each row has at least one non-zero entry.
+
+        threshold
+            Threshold or percentile depending on `mode`. If `mode` is `min_1`, `threshold` can be neglected.
+        batch_size
+            How many rows of the transport matrix to sparsify per batch.
+        n_samples
+            If `mode` is `percentile`, determine the number of samples based on which the percentile
+            is computed stochastically. Note this means that a matrix of shape `[n_samples, transport_matrix.shape[1]]`
+            has to be instantiated. If `None`, `n_samples` is set to batch_size.
+
+        Returns
+        -------
+        Nothing, but adds the sparsified transport matrix (:class:`scipy.sparse.csr_matrix`) to `self.sparsified_tmap`.
+
+        Note
+        ----
+        This function only serves for interfacing software which has to instantiate the transport matrix. Within moscot,
+        there is no point in using this function. We encourage users not to use this function unless it is necessary.
+        """
+        mode = SparsifyMode(mode)
+        if mode == SparsifyMode.THRESHOLD:
+            thr = threshold
+        elif mode == SparsifyMode.PERCENTILE:
+            n_samples = n_samples if n_samples is not None else batch_size
+            x = np.eye(self.shape[1], max(n_samples, self.shape[1]))
+            res = self.pull(x, scale_by_marginals=False)  # tmap @ indicator_vectors
+            thr = np.percentile(res, threshold)
+        elif mode == SparsifyMode.MIN_1:
+            thr = np.inf
+            for batch in range(0, self.shape[1], batch_size):
+                x = np.eye(self.shape[1], min(batch_size, self.shape[1] - batch), -(min(batch, self.shape[1])))
+                res = self.pull(x, scale_by_marginals=False)  # tmap @ indicator_vectors
+                thr_batch = res.max(axis=1).min()
+                thr = thr_batch if thr_batch < thr else thr
+            thr -= 1e-12  # necessary for the edge case
+        else:
+            raise NotImplementedError(mode)
+
+        tmaps_sparse: List[csr_matrix] = []
+        for batch in range(0, self.shape[1], batch_size):
```

Comment:
Small issue: if n << m, this is fairly inefficient. Would consider doing push in that case (and iterating over rows), since matmat operations are really efficient (and looping isn't).

Reply:
For the sake of reducing complexity I would prefer to keep it simple unless you have a strong opinion.

Reply:
It shouldn't be too complex, e.g.,:
```python
n, m = self.shape
if n < m
   k = n
   func = self.push
else:
   k = m
   func = self.pull

for i in range(0, k, batch_size)
   x = np.eye(...)
   res = func(x, scale_by_marginals=False)

Can you please add this?